### PR TITLE
Supply drops now use Drop Pods

### PR DESCRIPTION
**Pull request by fira**
## About The Pull Request

Supply drops now go through droppods instead of just appearing out of thin air.

USCM personnel is advised not to stand below the drop pods landing locations. Hopefully there is an ample warning of about 12 seconds to ensure this won't be an issue.

Refactored some code on the way and added a maptext notify to supply drops. You can also now fire OBs and Supply at Overwatch at the same time, in the offchance this ever matters.

Thanks to Jayne Hightower on Discord for the too-obvious-to-think-about yet simple and effective idea.

## Why It's Good For The Game

Visual feedback. Also visual feedback. But also, visual feedback.

Some would say immersion.

## Changelog

:cl:
add: Supply drops now use drop pods. USCM advises personnel not to stand below them.
qol: Squads will now get a text notification if a supply pad is launched for them, on top of just the chat one.
refactor: Refactored message sending code in squads/overwatch.
/:cl:

### CODE REVIEW

**Comment by hotimep** [comment]
marine drop pod related deaths about to mysteriously skyrocket

File: code/game/supplyshuttle.dm [hunk 1]
```diff
@@ -176,7 +175,7 @@ var/datum/controller/supply/supply_controller = new()
 	data["worldtime"] = world.time
 	data["x_offset"] = x_supply
 	data["y_offset"] = y_supply
-	data["active"] = busy
+	data["active"] = FALSE
```
Comment (harryob) [inline code/game/supplyshuttle.dm:178]:
```suggestion
```

for the interface:
```js
import { useBackend } from '../backend';
import {
  Button,
  Section,
  LabeledList,
  ProgressBar,
  Divider,
  NumberInput,
  NoticeBox,
  Box,
  Tabs,
} from '../components';
import { Window } from '../layouts';

export const SupplyDropConsole = (_props, context) => {
  const { act, data } = useBackend(context);

  const can_pick_squad = data.can_pick_squad;

  const timeLeft = data.nextfiretime - data.worldtime;
  const timeLeftPct = timeLeft / data.launch_cooldown;

  const cantFire = (timeLeft < 0, data.loaded === null);

  const squads = data.squad_list;

  return (
    <Window width={350} height={350}>
      <Window.Content scrollable>
        {!!can_pick_squad && (
          <NoticeBox info={1} fluid={1} textAlign="center">
            {data.current_squad
              ? `Current squad is :
                ${data.current_squad}`
              : 'No squad selected'}
          </NoticeBox>
        )}
        {!!can_pick_squad && (
          <Tabs horizontal textAlign="center" fluid>
            {squads.map((val) => {
              return (
                <Tabs.Tab
                  onClick={() => {
                    act('pick_squad', { squad_name: val.squad_name });
                  }}
                  key={val.squad_name}>
                  <Box color={val.squad_color}>{val.squad_name}</Box>
                </Tabs.Tab>
              );
            })}
          </Tabs>
        )}
        <Section title="Supply Drop">
          <LabeledList>
            <LabeledList.Item label="Longitude">
              <NumberInput
                width="4em"
                step={1}
                minValue={-1000}
                maxValue={1000}
                value={data.x_offset}
                onChange={(e, value) => act('set_x', { set_x: `${value}` })}
              />
            </LabeledList.Item>
            <LabeledList.Item label="Latitude">
              <NumberInput
                width="4em"
                step={1}
                minValue={-1000}
                maxValue={1000}
                value={data.y_offset}
                onChange={(e, value) => act('set_y', { set_y: `${value}` })}
              />
            </LabeledList.Item>
          </LabeledList>
          <Divider />
          <Section
            title="Supply Pad Status"
            buttons={
              <Button
                icon="sync-alt"
                content="Update"
                onClick={() => act('refresh_pad')}
              />
            }>
            <NoticeBox info={1} textAlign="center">
              {data.loaded
                ? `Supply Pad Status :
                  ${data.crate_name} loaded.`
                : 'No crate loaded.'}
            </NoticeBox>
            {(timeLeft < 0 && (
              <NoticeBox success={1} textAlign="center">
                Ready to fire!
              </NoticeBox>
            )) || (
              <ProgressBar
                width="100%"
                value={timeLeftPct}
                ranges={{
                  good: [-Infinity, 0.33],
                  average: [0.33, 0.67],
                  bad: [0.67, Infinity],
                }}>
                <Box textAlign="center">
                  {Math.ceil(timeLeft / 10)} seconds until next launch
                </Box>
              </ProgressBar>
            )}
            <Button
              fontSize="20px"
              textAlign="center"
              disabled={!!cantFire}
              fluid={1}
              icon="paper-plane"
              color="good"
              content="Launch Supply Drop"
              onClick={() => act('send_beacon')}
            />
          </Section>
        </Section>
      </Window.Content>
    </Window>
  );
};
```

**Review by harryob — APPROVED** [review]
thank you very cool, but

File: code/game/supplyshuttle.dm (hunk 1 above)
Comment (fira) [reply code/game/supplyshuttle.dm]: fixed

**Review by harryob — APPROVED** [review]

**Comment by AnuvKH** [comment]
Not sure if it's been mentioned but I don't think you can move the crate once it's been dropped. May be more important for lowpop Req crate scavengers.

**Pull request merged**